woocommerce/woocommerce-android
# Issue/6168 handling bug related to receipt email

Closes: #6168


### Description

The PR adds:
* Setting receipt emails when Stripe Plugin used
* Settings receipt email when WCPay version <4.0.0 is used

### Testing instructions

Stripe dashbaord can be used to verify that it works
* Try stripe payment, notice that receipt email is sent
* Try Wcpay payment, notice that receipt email is sent
* Try Wcpay payment with modified code:
```
val wcPayPluginInfo = wooStore.getSitePlugin(selectedSite.get(), WooCommerceStore.WooPlugin.WOO_PAYMENTS)
            ?.apply { version = "4.0.0" }
```
> CardReaderOnboardingChecker::95

notice that receipt email is not sent

<img width="430" alt="image" src="https://user-images.githubusercontent.com/4923871/162172920-81d08030-c1b4-4cf7-8ca3-cde46e6c30c0.png">
<img width="248" alt="image" src="https://user-images.githubusercontent.com/4923871/162173368-e8f3b301-7f59-4531-bbe2-4a3469d3b11f.png">


### Images/gif

No visual changes this PR brings
